server/utils/services/ratingServices.js

```diff
@@ -0,0 +1,24 @@
+
+const ratingServices = {
+  /**
+   * @param {array} ratingArray array of ratings
+   * @returns {Object} object contain rating detailRating
+   */
+  computeRating(ratingArray) {
+    let sumOfRating = 0;
+    const ratingCount = ratingArray.length;
+    ratingArray.forEach((rating) => {
```

Comment:
I thinkyou should remove this loop and use the SQL `AVE` function instead to calculate the average rating of the article.

Reply:
You can also use the `COUNT` function to get the number of ratings